core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -690,7 +690,7 @@ class LogManager(logDirs: Seq[File],
           if (preferredLogDir != null)
             List(new File(preferredLogDir))
           else
-            nextLogDirs()
+            nextLogDirs(topicPartition)
```

Comment:
The fact that this change didn't break any tests is concerning. Would it be easy to add some tests that verify the new behavior?

Reply:
> The fact that this change didn't break any tests is concerning. Would it be easy to add some tests that verify the new behavior?

Thanks for your attention, this strategy is to make decisions when creating a topic. The original strategy: For a broker, the disk to which the copy is stored is determined by the total number of partitions on the disk. Our strategy: It will give priority to whether the copy stored on the disk is even in the topic dimension, and then consider the number of partitions on the disk.
Can we reach a consensus: When creating a topic, if the number of partitions in the topic dimension is uniform on the disk, then the total number of partitions stored on each disk is even.
In addition, I tried to do some testing to further explain this. I will update to my jira.

Reply:
Personally I think the proposed strategy is better too, but we should be careful of its implications: if the total number of partitions across disks are not even at the beginning, then it will stay unbalanced; and only if the initial assignment of the partitions are even, then future added topics would still try to make it stay balanced with the total number of partitions. That means we may need to encourage some partition movement admin operations first before adding new topics.

Reply:
> Personally I think the proposed strategy is better too, but we should be careful of its implications: if the total number of partitions across disks are not even at the beginning, then it will stay unbalanced; and only if the initial assignment of the partitions are even, then future added topics would still try to make it stay balanced with the total number of partitions. That means we may need to encourage some partition movement admin operations first before adding new topics.

I will design some test cases to test the problems you are worried about.